### PR TITLE
[13.0][FIX] web_drop_target: Don't collide with odoo drag and drop implementation

### PR DESCRIPTION
**Pull request by Tardo**
+ Old:
dg_old

+ New:
dg_new

cc @tecnativa TT32093

### CODE REVIEW

File: web_drop_target/README.rst [hunk 1]
```diff
@@ -14,13 +14,13 @@ Drop target support
     :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
     :alt: License: AGPL-3
 .. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fweb-lightgray.png?logo=github
-    :target: https://github.com/OCA/web/tree/13.0/web_drop_target
+    :target: https://github.com/OCA/web/tree/v13/web_drop_target
```
Comment (pedrobaeza) [inline web_drop_target/README.rst]: You have used an incorrect version while generating the README

File: web_drop_target/static/src/js/web_drop_target.js [hunk 2]
```diff
@@ -1,84 +1,221 @@
...
-            e.preventDefault();
+        /**
+         * TODO: Change name to follow the standard in new versions
+         *
+         * @private
+         */
+        _add_overlay: function() {
+            if (!this._drop_overlay || this._drop_overlay.length === 0) {
+                this.$drop_zone = this.$(this._drop_zone_selector).last();
+                // Element that represents the zone where can drop files
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
                // Element that represents the zone where you can drop files
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 3]
```diff
@@ -1,84 +1,221 @@
...
+                // TODO: This name is preserved to don't break the compatibility with other modules,
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
                // TODO: This name is preserved for not breaking the compatibility with other modules,
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 4]
```diff
@@ -1,84 +1,221 @@
...
+                }
+                this._drop_overlay.css(overlay_css);
+                $("body").append(this._drop_overlay);
+
+                this._drop_overlay.on("drop", this._on_drop.bind(this));
+            }
+        },
+
+        /**
+         * TODO: Change name to follow the standard in new versions
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
         * TODO: Change the name to follow the standard in new versions
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 5]
```diff
@@ -1,84 +1,221 @@
...
+         * @private
+         * @param {MouseEvent} ev
+         */
+        _onBodyFileDrop: function(ev) {
+            ev.preventDefault();
+            $(".o_drag_over").addClass("d-none");
+        },
+
+        /**
+         * TODO: Change name to follow the standard in new versions
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
         * TODO: Change the name to follow the standard in new versions
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 6]
```diff
@@ -1,84 +1,221 @@
...
+            ev.preventDefault();
+            const drop_items = this._get_drop_items(ev);
+            if (_.isEmpty(drop_items)) {
+                return;
+            }
+            this._handle_drop_items(drop_items, ev);
+        },
+
+        /**
+         * TODO: Change name to follow the standard in new versions
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
         * TODO: Change the name to follow the standard in new versions
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 7]
```diff
@@ -1,84 +1,221 @@
...
+                        _.contains(this._drop_allowed_types, item.type)
+                    );
+                }
             }
-            return null;
+            return drop_items;
         },
 
+        /**
+         * TODO: Change name to follow the standard in new versions
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
         * TODO: Change the name to follow the standard in new versions
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 8]
```diff
@@ -1,84 +1,221 @@
...
-        _handle_drop_items: function(drop_items, e) {
+        _handle_drop_items: function(drop_items, ev) {
             // Do something here, for example call the helper function below
             // e is the on_load_end handler for the FileReader above,
             // so e.target.result contains an ArrayBuffer of the data
         },
 
-        _create_attachment: function(file, reader, e, res_model, res_id, extra_data) {
+        /**
+         * TODO: Change name to follow the standard in new versions
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
         * TODO: Change the name to follow the standard in new versions
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 9]
```diff
@@ -92,100 +229,69 @@ odoo.define("web_drop_target", function(require) {
...
+                    res.chatter.trigger_up("reload");
+                    res.chatter.$(".o_chatter_button_attachment").click();
                 }
             });
         },
 
-        _file_reader_error_handler: function(e) {
-            console.error(e);
+        /**
+         * TODO: Change name to follow the standard in new versions
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
         * TODO: Change the name to follow the standard in new versions
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 10]
```diff
@@ -92,100 +229,69 @@ odoo.define("web_drop_target", function(require) {
...
+        _file_reader_error_handler: function(ev) {
+            console.error(ev);
         },
 
-        _handle_file_drop_attach: function(item, e, res_model, res_id, extra_data) {
-            var self = this;
-            var file = item;
-            if (!file || !(file instanceof Blob)) {
+        /**
+         * TODO: Change name to follow the standard in new versions
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js]:
```suggestion
         * TODO: Change the name to follow the standard in new versions
```

File: web_drop_target/static/src/js/web_drop_target.js [hunk 11]
```diff
@@ -1,84 +1,221 @@
...
+        /**
+         * @private
+         * @param {MouseEvent} ev
+         */
+        _onBodyFileDragLeave: function(ev) {
+            // On every dragenter chain created with parent child element
+            // That's why dragleave is fired every time when a child elemnt is hovered
+            // so here we hide dropzone based on mouse position
+            if (
+                ev.originalEvent.clientX <= 0 ||
```
Comment (pedrobaeza) [inline web_drop_target/static/src/js/web_drop_target.js:126]: Is this taking into account non maximized windows that are in the middle of the screen?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
On a functional try, I have discovered these 2 things:

* On non maximized windows, the placeholder appears incorrect:
Selección_016

* When having the message popup (or in general a Summernote editor), the placeholder is avoiding to drag and drop on this area:
Selección_017

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
Nice refactor. Works like charm :+1:

**Comment by pedrobaeza** [comment]
/ocabot merge patch

Please forward port it to v14

**Pull request merged**